libs/datasets/data_source.py

```diff
@@ -68,6 +70,24 @@ def population(self) -> LatestValuesDataset:
         """Builds generic beds dataset"""
         return self.latest_values()
 
+    @lru_cache(None)
+    def multi_region_dataset(self) -> MultiRegionDataset:
+        if set(self.INDEX_FIELD_MAP.keys()) == set(TimeseriesDataset.INDEX_FIELDS):
+            dataset = MultiRegionDataset.from_fips_timeseries_df(self.data).add_provenance_all(
```

Comment:
how do you feel about adding a bit more functionality to `from_fips_timeseries_df`, or maybe:
```
def from_fips_df(timeseries=None, static=None, provenance=None):
   ...

```

instead of needing all of the `add_*` functions

Reply:
I had something like that originally but moved towards a builder-like API because it helped with factoring out the independent parts of the static 'constructor' methods. It'd be nice to try to streamline how this things are built but not today.